Tools/CMake/AMReXOptions.cmake

```diff
@@ -0,0 +1,323 @@
+###############################################
+
+# Here we define the default config options   #
+# that can be overwritten by the user         #
+
+###############################################
+include_guard(GLOBAL)
+
+include(CMakeDependentOption)
+
+#
+# Define a macro to print active options
+#
+macro (print_option _var)
+   if (${_var})
+      message( STATUS "   ${_var}")
+   endif ()
+endmacro ()
+
+
+#
+# Dimensionality of the build  ===============================================
+#
+set(AMReX_SPACEDIM_VALUES 1 2 3)
+set(AMReX_SPACEDIM 3 CACHE STRING "Dimension of AMReX build: <1,2,3>")
+set_property(CACHE AMReX_SPACEDIM PROPERTY STRINGS ${AMReX_SPACEDIM_VALUES})
+if(NOT AMReX_SPACEDIM IN_LIST AMReX_SPACEDIM_VALUES)
+   message(FATAL_ERROR "AMReX_SPACEDIM=${AMReX_SPACEDIM} is not allowed."
+      " Must be one of ${AMReX_SPACEDIM_VALUES}")
+endif()
+message( STATUS "Building AMReX with AMReX_SPACEDIM = ${AMReX_SPACEDIM}")
+
+#
+# Populate the cache and check the value of the user-definable options
+#
+message(STATUS "Configuring AMReX with the following options enabled: ")
+
+#
+# This is the option to enable/disable xSDK mode
+#
+# To handle both XDSK options and correponding plain AMReX options,
+# we make use of policy CMP0077 introduced as a default in CMake 3.13
+# Under policy CMP0077, normal variables prevent option()
+# to set internal variables of the same name.
+# Example: if XSDK mode is ON and XSDK_ENABLE_Fortran=ON, a normal
+# variable AMReX_FORTRAN will be created and set to ON.
+# This will stop the subsequent option( AMReX_FORTRAN "Enable Fortran language" ON )
+# from being executed and no entry AMReX_FORTRAN will be created in the cache
+#
+option( USE_XSDK_DEFAULTS "Enable xSDK mode"  OFF )
```

Comment:
It would be pretty nice if we could fulfill the xSDK policies by default for high usability, also because it would be less logic to test.
Do you and CC @WeiqunZhang see a way towards that? Maybe this is a good time to introduce this as well, just curious.

Reply:
CMake: should this be pre-fixed/renamed to `AMReX_XSDK`?